javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -0,0 +1,826 @@
+        return ofAll(comparator, toList().replace(currentElement, newElement));
+    }
+
+    @Override
+    public PriorityQueue<T> replaceAll(T currentElement, T newElement) {
+        Objects.requireNonNull(currentElement, "currentElement is null");
+        Objects.requireNonNull(newElement, "newElement is null");
+        return ofAll(comparator, toList().replaceAll(currentElement, newElement));
+    }
+
+    @Override
+    public PriorityQueue<T> retainAll(Iterable<? extends T> elements) {
+        return Collections.retainAll(this, elements);
+    }
+
+    public PriorityQueue<T> removeAll(Iterable<? extends T> elements) {
+        return Collections.removeAll(this, elements);
+    }
+
+    @Override
+    public PriorityQueue<T> scan(T zero, BiFunction<? super T, ? super T, ? extends T> operation) {
+        return Collections.scanLeft(this, zero, operation, PriorityQueue.empty(comparator), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public <U> PriorityQueue<U> scanLeft(U zero, BiFunction<? super U, ? super T, ? extends U> operation) {
+        Objects.requireNonNull(operation, "operation is null");
+        return Collections.scanLeft(this, zero, operation, empty(naturalComparator()), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public <U> PriorityQueue<U> scanRight(U zero, BiFunction<? super T, ? super U, ? extends U> operation) {
+        Objects.requireNonNull(operation, "operation is null");
+        return Collections.scanRight(this, zero, operation, empty(naturalComparator()), PriorityQueue::enqueue, Function.identity());
+    }
+
+    @Override
+    public Iterator<? extends PriorityQueue<T>> sliding(long size) {
+        return iterator().sliding(size).map(v -> ofAll(comparator, v));
+    }
+
+    @Override
+    public Iterator<? extends PriorityQueue<T>> sliding(long size, long step) {
+        return iterator().sliding(size, step).map(v -> ofAll(comparator, v));
+    }
+
+    @Override
+    public Tuple2<? extends PriorityQueue<T>, ? extends PriorityQueue<T>> span(Predicate<? super T> predicate) {
+        Objects.requireNonNull(predicate, "predicate is null");
+        return Tuple.of(takeWhile(predicate), dropWhile(predicate));
+    }
+
+    @Override
+    public Option<? extends PriorityQueue<T>> tailOption() {
+        return isEmpty() ? Option.none() : Option.some(tail());
+    }
+
+    @Override
+    public PriorityQueue<T> take(long n) {
+        if (n >= length()) {
+            return this;
+        } else {
+            PriorityQueue<T> source = this, result = empty(comparator);
+            for (int i = 0; i < n; i++) {
+                result = result.enqueue(source.head());
+                source = source.tail();
+            }
+            return result;
+        }
+    }
+
+    @Override
+    public PriorityQueue<T> takeRight(long n) {
+        return ofAll(comparator, toList().takeRight(n));
```

Comment:
I believe you should use <code>iterator()</code> instead of <code>toList()</code> here.
There is no difference in behavior but <code>toList()</code> doing unnecessary <code>List.reverse()</code> call